model_zoo/mnist/mnist_pytorch.py

```diff
@@ -131,15 +107,14 @@ def train(args):
     train_data = torchvision.datasets.ImageFolder(args.training_data)
     test_data = torchvision.datasets.ImageFolder(args.validation_data)
 
-    allreduce_controller = create_elastic_controller(
+    data_shard_service = RecordIndexService(
         batch_size=args.batch_size,
         dataset_size=len(train_data.imgs),
         num_epochs=args.num_epochs,
         shuffle=True,
+        dataset_name="mnist_training_data",
     )
-    train_dataset = ElasticDataset(
-        train_data.imgs, allreduce_controller.data_shard_service
-    )
+    train_dataset = ElasticDataset(train_data.imgs, data_shard_service)
```

Comment:
Does the 'dataset' here refer to some data sets obtained by the worker through shards index instead of all training data sets? Because I think the traing dataset size is fixed.

L49:   # Set the maxsize because the size of dataset is not fixed
            # when using dynamic sharding

Reply:
Yee, the dataset read images by sample indices in the shard.